### DISCUSSION TITLE
I want to use ffmpeg component ffplay.exe as a player, combined with my UI to execute ffplay.exe command, how to achieve

### DISCUSSION THREAD

**Discussion by muyaCode** [tauri-apps/tauri#8580] [Q&A]
I am puzzled about how to integrate the video playback executed by the ffplay.exe binary with my frontend UI.

My implementation approach is as follows:

Embedding the ffplay binary: Embed the ffplay binary file into the Tauri application. Tauri supports the embedding of external binary files like ffplay.exe through configuration.

Invoking ffplay execution command: Utilize Tauri's frontend library shell's Command method to call the embedded ffplay binary and execute the playback command in Vue3. However, when playing, it opens the ffplay window separately. I aim to integrate ffplay with my frontend UI to execute ffplay commands.

UI integration: Create a video playback area in Vue3's UI, which can be a video element or other suitable UI components. Ensure this area integrates well with Tauri's frontend library.

Passing the video file path: If playing a specific video file is necessary, ensure to pass the correct video file path when calling the ffplay command. This can be achieved by setting a file path variable in Vue3 and referencing it in the shell command.

Handling playback controls: Implement playback controls such as pause, stop, etc. Ensure your UI can communicate with Tauri's frontend library and pass the corresponding parameters when calling the ffplay command.

Handling real-time video streaming (if needed): If your application requires real-time video streaming, ensure Tauri and ffplay can adapt to streaming playback. Additional settings and configurations may be needed to handle video streaming.

**Comment by hanaTsuk1** [comment]
In a word, you just want to call Command without displaying the window, right?
```
// https://learn.microsoft.com/zh-cn/windows/win32/procthread/process-creation-flags
Command::new(command).creation_flags(0x08000000);  // CREATE_NO_WINDOW
```

**Reply by muyaCode** [reply]
No, it's used as a player with the GUI configuration I wrote
